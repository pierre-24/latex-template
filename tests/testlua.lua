-- find the packages:
package.path = '../utf8.lua/?.lua;../?.lua;' .. package.path
zmd = require 'zmdocument'

-- run test on authors
print('% --- Test authors')
print('% one author:')
print(zmd.formatAuthorsBase('Eskimon', '', false))

print('% three authors:')
print(zmd.formatAuthorsBase('Karnaj, Heziode, pierre_24', '', false))

print('% one author with link:')
print(zmd.formatAuthorsBase('Eskimon', 'http://test.com/author', false))

print('% one author with long name:')
print(zmd.formatAuthorsBase('that_author_with_a_very_long_name', '', false))

print('% one author with long name (but in author list, so full name):')
print(zmd.formatAuthorsBase('that_author_with_a_very_long_name', '', true))

-- run test on abbrv
print('% --- Test abbrv')
print(zmd.cleanSpecialCharacters('AT\\&T'))
print(zmd.cleanSpecialCharacters('pierre\\_24'))
print(zmd.cleanSpecialCharacters('\\#hashtag'))
