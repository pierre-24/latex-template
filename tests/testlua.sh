#!/bin/bash
expected=$(cat ./testlua.lua.expected)
actual=$(texlua ./testlua.lua)
if [[ "$expected" != "$actual" ]]; then 
    echo '!! The output of `texlua testlua.lua` changed: zmdocument.lua was modified:'
    diff  <(echo "$actual" ) <(echo "$expected")
    echo '!! Did you forget to do `texlua ./testlua.lua > testlua.lua.expected` ?'
    exit 1
else
    echo 'zmdocument.lua was not modified.'
fi
