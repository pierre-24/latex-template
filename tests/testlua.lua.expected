% --- Test authors
% one author:
\href{/Eskimon/}{\color{titlePageAuthorColor}\bsc{Eskimon}}\\
% three authors:
\href{/Karnaj/}{\color{titlePageAuthorColor}\bsc{Karnaj}}\\,\href{/Heziode/}{\color{titlePageAuthorColor}\bsc{Heziode}}\\,\href{/pierre_24/}{\color{titlePageAuthorColor}\bsc{pierre_24}}\\
% one author with link:
\href{http://test.com/author/Eskimon/}{\color{titlePageAuthorColor}\bsc{Eskimon}}\\
% one author with long name:
\href{/that_author_with_a_very_long_name/}{\color{titlePageAuthorColor}\bsc{that_author_...}}\\
% one author with long name (but in author list, so full name):
\href{/that_author_with_a_very_long_name/}{\color{titlePageAuthorColor}\bsc{that_author_with_a_very_long_name}}\\
% --- Test abbrv
AT(amp)T
pierre(und)24
(num)hashtag
